choose: add page (#6135)
tldr-pages/tldr: pages/common/choose.md

# choose

> A human-friendly and fast alternative to cut and (sometimes) awk
> More information: <https://github.com/theryangeary/choose>.

> - Print the 5th item from a line (zero indexed):
> `choose 5`

Reviewer: ```suggestion
- Print the Nth item from a line (zero indexed):
`choose {{number}}`
```

> - Print the 5th item from a line (zero indexed):
> `choose 5`

Reviewer 2: ```suggestion
- Print the 5th item from a line (zero indexed):

`choose {{5}}`
```

- Print the 0th, 3rd, and 5th item from a line, where items are separated by ':' instead of whitespace:

> `choose -f ':' 0 3 5`

Reviewer 2: ```suggestion
`choose -f '{{:}}' {{0}} {{3}} {{5}}`
```

> - Print the 0th, 3rd, and 5th item from a line, where items are separated by ':' instead of whitespace:
> 
> `choose -f ':' 0 3 5`
> 

Reviewer: ```suggestion
- print the specified nth items from a line, where items are separated by ':' instead of whitespace:

`choose -f ':' 0 3 5`

```

> - Print everything from the 2nd to 5th item on the line, inclusive of the 5th:
> 
> `choose 2:5`

Reviewer 2: ```suggestion
- Print everything from the 2nd to 5th item on the line, including the 5th:

`choose {{2}}:{{5}}`
```

> - print everything from the 2nd to 5th item on the line, exclusive of the 5th:
> 
> `choose -x 2:5`

Reviewer 2: ```suggestion
- print everything from the 2nd to 5th item on the line, excluding the 5th:

`choose -x {{2}}:{{5}}`
```

> - Print the beginning of the line to the 3rd item:
> 
> `choose :3`

Reviewer 2: ```suggestion
- Print the beginning of the line to the 3rd item:

`choose :{{3}}`
```

> - print the beginning of the line to the 3rd item, exclusive:
> 
> `choose -x :3`

Reviewer 2: What do you think about this slight change of wording?
```suggestion
- Print all items from the beginning of the line until the 3rd item (exclusive):

`choose -x :{{3}}`
```

> - Print the third item to the end of the line:
> 
> `choose 3:`

Reviewer 2: ```suggestion
- Print all items from the 3rd to the end of the line:

`choose {{3}}:`
```

- Print the last item from a line

> `choose -1`

Reviewer 2: ```suggestion
`choose {{-1}}`
```

# choose

> > A human-friendly and fast alternative to cut and (sometimes) awk.

Reviewer 3: I think `user-friendly` is more common, but `human-friendly` is also an interesting alternative. 😄

Reviewer 4: Not for bots! 🤖😉

Reviewer: @tldr-bot thoughts?

Reviewer 5: I prefer `user-friendly`, because I'm not a human but the command itself is friendly to me :-)

> More information: <https://github.com/theryangeary/choose>.

> - Print the 5th item from a line (zero indexed):

Reviewer 3: ```suggestion
- Print the 5th item from a line (starting from 0):
```
I think `zero-indexed` is sounds a bit too complicated.

`choose {{5}}`

> - Print the 0th, 3rd, and 5th item from a line, where items are separated by ':' instead of whitespace:

Reviewer 3: I'm not sure about the way you use `Nth`. To me, the first element is at endex 0 and there is no 0th element. Any opinions on this @bl-ue, @CleanMachine1?

Reviewer 6: Depending on the underlying programming language and convention @marchersimon, things can start at 0 rather than 1.

Reviewer 3: Makes sense. But regardless of that, as @ SethFalco pointed out on Gitter (I have no idea how to copy the link to that thread), the element at the first position of a list is the first element, regardless of the indexing.

> `choose -f '{{:}}' {{0}} {{3}} {{5}}`

Reviewer 3: ```suggestion
`choose --field-separator '{{:}}' {{0}} {{3}} {{5}}`
```
We always use long options when possible.

> `choose --field-separator '{{:}}' {{0}} {{3}} {{5}}`

Reviewer: Maybe the rest should be converted into words, third and fifth

- Print everything from the 2nd to 5th item on the line, including the 5th:

> `choose {{2}}:{{5}}`

Reviewer 3: ```suggestion
`choose {{1}}:{{4}}`
```

- Print everything from the 2nd to 5th item on the line, excluding the 5th:

> `choose -x {{2}}:{{5}}`

Reviewer 3: ```suggestion
`choose --exclusive {{2}}:{{5}}`
```

> `choose --exclusive {{2}}:{{5}}`

Reviewer 3: ```suggestion
`choose --exclusive {{1}}:{{4}}`
```

- Print the beginning of the line to the 3rd item:

> `choose :{{3}}`

Reviewer 3: ```suggestion
`choose :{{2}}`
```

- Print all items from the beginning of the line until the 3rd item (exclusive):

> `choose -x :{{3}}`

Reviewer 3: ```suggestion
`choose --exclusive :{{3}}`
```

> `choose --exclusive :{{3}}`

Reviewer 3: ```suggestion
`choose --exclusive :{{2}}`
```

- Print all items from the 3rd to the end of the line:

> `choose {{3}}:`

Reviewer 3: ```suggestion
`choose {{2}}:`
```

Reviewer: Does it start from 0?

Reviewer 3: It does.

Reviewer: Then these changes need to be made.
